File: TypeScript/benchmark.ts
#!/usr/local/bin/tsc

import Fft from './fft';

const LOG2FFTSIZE: number = 12;
const FFT_REPEAT: number = 1000;

function main() {
    const SIZE: number = 1 << LOG2FFTSIZE;
    const xy: number[][] = [];
    let xy_out: number[][] = [];    
    
    for (var i = 0; i < SIZE / 2; i++) xy[i] = [1., 0];
    for (; i < SIZE; i++) xy[i] = [-1., 0];
    const fft = new Fft();

    // FFT
    const starttime: Date = new Date();

    for (let i = 0; i < FFT_REPEAT; i++) xy_out = fft.calculate(LOG2FFTSIZE, xy);

    const endtime: Date = new Date();
    const eltime = endtime.getTime() - starttime.getTime(); // msec
    console.log("\n--> " + FFT_REPEAT + " piece of " + (1 << LOG2FFTSIZE) + " pt FFT;  " + (eltime / FFT_REPEAT) + " ms/piece");

    for (let i = 0; i < 6; i++) {
        console.log(i + "\t" + xy_out[i][0] + "\t" + xy_out[i][1]);
    }

    return 0;
}

main();

File: Dart/bin/benchmark.dart
import 'package:benchmark/fft.dart';
import 'package:complex/complex.dart';

const int log2FftSize = 12;
const int fftRepeat = 10000;
const size = 1 << log2FftSize;

void main(List<String> arguments) {
  int i;

  var xy = List<Complex>.filled(size, Complex(1.0, 0.0), growable: false);
  var xyOut =
      List<Complex>.filled(xy.length, Complex(0.0, 0.0), growable: false);

  for (var i = size >> 1; i < size; i++) {
    xy[i] = Complex(-1.0, 0.0);
  }

  final fft = Fft();

  // JIT warm up ... possible give more speed
  for (i = 0; i < fftRepeat; i++) {
    fft.calculate(log2FftSize, xy, xyOut);
  }

  // FFT
  var stopwatch = Stopwatch();
  stopwatch.start();

  for (i = 0; i < fftRepeat; i++) {
    fft.calculate(log2FftSize, xy, xyOut);
  }

  stopwatch.stop();

  print('Total ($fftRepeat): ${stopwatch.elapsedMilliseconds}');

  final tpp = stopwatch.elapsedMilliseconds / fftRepeat;

  print('$fftRepeat piece(s) of ${1 << log2FftSize} pt FFT; $tpp ms/piece\n');

  for (i = 0; i < 6; i++) {
    print('$i\t${xyOut[i]}');
  }
}
